### PR TITLE
feat(auth): implement auto-login with saved credentials (SEC-6.9)

### PR DESCRIPTION
**Pull request by pcamarajr**
## Summary

This PR implements automatic credential loading on app startup and auto-login functionality. When the app starts, it checks for saved credentials and automatically attempts registration if credentials are found.

## Changes

- Update `auth_service.rs` to save 'default_account' pointer when saving credentials
- Add `load_saved_credentials` Tauri command to retrieve saved credentials
- Add `loadSavedCredentials()` and `autoLogin()` methods to `authStore.ts`
- Trigger auto-login on app startup in login page
- Update documentation to mark SEC-6.9 as completed

### How to Test

1. **Setup**:
   ```bash
   git checkout feat/load-credentials-on-startup-auto-login
   npm install
   ```

2. **Test Steps**:
   - Start the app and register with valid SIP credentials
   - Close the app completely
   - Restart the app and navigate to the login page
   - Verify that auto-login is attempted automatically
   - Verify that registration succeeds without manual input
   - Verify that the app redirects to the dialer after successful auto-login

3. **Expected Behavior**:
   - On first launch (no saved credentials): Login form should be shown
   - After successful registration: Credentials should be saved
   - On subsequent launches: Auto-login should be attempted automatically
   - If auto-login fails: Login form should be shown for manual entry

### Test Coverage

- [x] Manual testing completed
- [x] Edge cases tested (no credentials, invalid credentials)

### Code Quality

- [x] Code follows project conventions
- [x] Functions are focused and single-purpose
- [x] Variable names are descriptive
- [x] No code duplication
- [x] Error handling is appropriate

### Framework Compliance

- [x] Svelte 5 runes mode (no legacy syntax)
- [x] Rust code follows idiomatic patterns
- [x] Tauri IPC follows best practices
- [x] Design system components used correctly

### Security

- [x] No hardcoded secrets
- [x] Input validation present
- [x] Sensitive data handled securely
- [x] Tauri capabilities properly configured

### Documentation

- [x] README.md updated
- [x] Code comments explain complex logic
- [x] Architecture docs updated

## Breaking Changes

None

## Related Issues

Implements SEC-6.9 from the implementation roadmap.

## Additional Notes

- Uses a 'default_account' pointer approach since the credential store doesn't have a 'list all' method
- Auto-login gracefully handles failures and falls back to manual login
- Credentials are stored securely in the platform keychain (macOS Keychain)

### CODE REVIEW

File: src-tauri/src/services/auth_service.rs [hunk 1]
```diff
@@ -180,6 +180,30 @@ impl AuthService {
                             match store.save(&key_clone, &creds_clone).await {
                                 Ok(()) => {
                                     eprintln!("DEBUG:[AUTH_SERVICE/SAVE_CREDENTIALS] Credentials saved successfully");
+
+                                    // Also save the default_account pointer
```
Comment (pcamarajr) [inline src-tauri/src/services/auth_service.rs]: **⚠️ Warning:** Code duplication detected

The default_account pointer saving logic (lines 184-206) is duplicated at lines 214-234. Consider extracting this into a helper function to follow DRY principles.

**Suggestion:**
```rust
impl AuthService {
    async fn save_default_account_pointer(
        store: &Arc<dyn CredentialStore>,
        credential_key: &str,
    ) {
        // ... pointer saving logic ...
    }
}
```

This would reduce code duplication and make maintenance easier.

File: src/lib/stores/authStore.ts [hunk 2]
```diff
@@ -231,5 +231,75 @@ export const authStore = {
...
+      console.log("DEBUG:[AUTHSTORE/AUTO_LOGIN] Found saved credentials, attempting registration", {
+        server: credentials.server,
+        username: credentials.username,
+      });
+      
+      // Attempt registration with saved credentials
+      await authStore.register(
+        credentials.server,
+        parseInt(credentials.port, 10),
+        credentials.protocol.toLowerCase() as "udp" | "tcp" | "tls",
```
Comment (pcamarajr) [inline src/lib/stores/authStore.ts]: **💡 Suggestion:** Protocol conversion type safety

The protocol conversion uses `toLowerCase()` with a type assertion. While this works for the expected values ("Udp", "Tcp", "Tls"), consider adding validation for robustness:

```typescript
const protocolMap: Record<string, "udp" | "tcp" | "tls"> = {
  "Udp": "udp",
  "Tcp": "tcp",
  "Tls": "tls",
};

const protocol = protocolMap[credentials.protocol] || credentials.protocol.toLowerCase() as "udp" | "tcp" | "tls";
```

This provides better type safety and handles unexpected protocol values more gracefully.

**Review by pcamarajr — COMMENTED** [review]
## Code Review Summary

✅ **Overall Assessment:** The PR successfully implements auto-login functionality with saved credentials. The implementation follows the architecture patterns and correctly updates documentation. Minor code quality improvements are suggested.

**Issues Found:**
- 🚫 Blockers: 0
- ⚠️ Warnings: 1
- 💡 Suggestions: 1

**Key Findings:**

1. **Code Duplication** (`src-tauri/src/services/auth_service.rs:184`): The default_account pointer saving logic is duplicated in two places (after initial registration and after refresh). Consider extracting to a helper function.

2. **Protocol Conversion Type Safety** (`src/lib/stores/authStore.ts:292`): The protocol conversion could be more robust with explicit mapping instead of relying solely on `toLowerCase()`.

**Review Details:**

### ✅ Functionality & Correctness
- Auto-login flow correctly loads saved credentials and attempts registration
- Error handling is appropriate with graceful fallback to manual login
- Edge cases handled (no credentials, invalid credentials)

### ✅ Code Quality & Maintainability
- Code is readable and well-structured
- Functions are focused and single-purpose
- Variable names are descriptive
- **Minor issue**: Code duplication in default_account pointer saving (see inline comment)

### ✅ Security
- No hardcoded secrets
- Credentials stored securely in platform keychain
- Sensitive data handled appropriately
- Tauri capabilities properly configured

### ✅ Architecture Compliance
- Follows Clean Architecture principles
- Layer boundaries respected
- Business logic in correct layer
- Platform-specific code properly abstracted

### ✅ Svelte 5 Framework Compliance
- Uses Svelte 5 runes mode syntax (`$effect`, `onMount`)
- No legacy `$:` reactive statements
- No legacy `on:click` syntax
- Event handlers properly typed
- Store subscriptions work correctly with legacy stores

### ✅ Rust & Tauri Compliance
- Code compiles without warnings
- Follows Rust ownership and borrowing rules
- Error handling uses `Result<T, E>` appropriately
- Tauri commands use proper error types
- Async code uses Tokio runtime correctly

### ✅ Design System & UI Compliance
- Uses appropriate components
- Follows design system patterns
- No UI changes in this PR

### ✅ Testing & Documentation
- README.md correctly updated to reflect SEC-6.9 completion
- Architecture documentation updated
- Code comments explain the default_account pointer approach
- Manual testing completed per PR description

### ✅ Performance & Optimization
- Async operations don't block UI thread
- Credential loading is efficient
- No obvious performance bottlenecks

### ✅ README & Documentation Updates
- README.md correctly updated with SEC-6.9 completion status
- Architecture roadmap updated with implementation details
- Documentation accurately reflects current state

**Categories Reviewed:**
- Functionality & Correctness ✅
- Code Quality & Maintainability ⚠️ (minor duplication)
- Security ✅
- Architecture Compliance ✅
- Svelte 5 Framework Compliance ✅
- Rust & Tauri Compliance ✅
- Design System & UI Compliance ✅
- Testing & Documentation ✅
- Performance & Optimization ✅
- README & Documentation Updates ✅

**Recommendation:** Approve with minor suggestions. The code duplication is a maintainability concern but doesn't block the PR. The protocol conversion suggestion is a nice-to-have improvement.

File: src/lib/stores/authStore.ts (hunk 2 above)
Comment (pcamarajr) [reply src/lib/stores/authStore.ts]: ✅ Resolved in commit 4855af18f6dfdb86192d6f7a7285e931a9c0c2c3
